Source/WebCore/Modules/url-pattern/URLPatternParser.cpp

```diff
@@ -330,5 +329,219 @@ String escapeRegexString(StringView input)
     return escapeRegexStringForCharacters(input.span16());
 }
 
+// https://urlpattern.spec.whatwg.org/#convert-a-modifier-to-a-string
+String convertModifierToString(Modifier modifier)
+{
+    switch (modifier) {
+    case Modifier::ZeroOrMore:
+        return "*"_s;
+    case Modifier::Optional:
+        return "?"_s;
+    case Modifier::OneOrMore:
+        return "+"_s;
+    default:
+        return { };
+    }
+}
+
+// https://urlpattern.spec.whatwg.org/#generate-a-regular-expression-and-name-list
+std::pair<String, Vector<String>> generateRegexAndNameList(const Vector<Part>& partList, const URLPatternStringOptions& options)
+{
+    StringBuilder result;
+    result.append("^"_s);
+
+    Vector<String> nameList;
+
+    for (const Part& part : partList) {
+        if (part.type == PartType::FixedText) {
+            if (part.modifier == Modifier::None)
+                result.append(escapeRegexString(part.value));
+            else
+                result.append(makeString("(?:"_s, escapeRegexString(part.value), ")"_s, convertModifierToString(part.modifier)));
```

Comment:
It is inefficient to construct a temporary string just to append to a StringBuilder. I think you can write it more efficiently like so:
```
result.append("(?:"_s, escapeRegexString(part.value), ")"_s, convertModifierToString(part.modifier));
```
If that doesn't work, there's always:
```
result.append("(?:"_s);
result.append(escapeRegexString(part.value));
...
```

Reply:
The same comment likely applies in other places.